pkg/reconciler/taskrun/taskrun.go

```diff
@@ -497,16 +505,13 @@ func (c *Reconciler) updateLabelsAndAnnotations(ctx context.Context, tr *v1beta1
 func (c *Reconciler) handlePodCreationError(ctx context.Context, tr *v1beta1.TaskRun, err error) error {
 	switch {
 	case isExceededResourceQuotaError(err):
-		backoff, currentlyBackingOff := c.timeoutHandler.GetBackoff(tr.GetNamespacedName(), *tr.Status.StartTime, *tr.Spec.Timeout)
-		if !currentlyBackingOff {
-			go c.timeoutHandler.SetTimer(tr.GetNamespacedName(), time.Until(backoff.NextAttempt))
-		}
-		msg := fmt.Sprintf("TaskRun Pod exceeded available resources, reattempted %d times", backoff.NumAttempts)
```

Comment:
This is another change worth calling out.  Previously this bounded the number of attempts, but this state was stored in-memory in the controller process, so it would be lost across failovers and rollouts.

We could perhaps add logic here to `snooze` this, but I went with the simplest change as a starting point for discussion.

Reply:
Will exceeding resources be retried later until pod creation succeeds? It seems like a regression to just immediately fail in that case.

Reply:
This returns the error so it will be requeued without delay, and the workqueue itself should backoff.